lib/Parser/Parse.cpp

```diff
@@ -5054,9 +5054,23 @@ bool Parser::ParseFncDeclHelper(ParseNodePtr pnodeFnc, ParseNodePtr pnodeFncPare
             {
                 OUTPUT_TRACE_DEBUGONLY(Js::ParsePhase, _u("The param and body scope of the function %s cannot be merged\n"), pnodeFnc->sxFnc.pnodeName ? pnodeFnc->sxFnc.pnodeName->sxVar.pid->Psz() : _u("Anonymous function"));
                 // Add a new symbol reference for each formal in the param scope to the body scope.
-                paramScope->ForEachSymbol([this](Symbol* param) {
+                // Retain the same position of formals in the body variables too.
+                paramScope->ForEachSymbol([this, pnodeFnc](Symbol* param) {
```

Comment:
While inserting the symbols to the symbol list we always insert at the front as don't have the tail pointer. Changing this behavior will require changing the symbol list implementation. I will skip that for now.